src/piecewise.jl

```diff
@@ -0,0 +1,208 @@
+struct Domain{L, R, T, S}
+    lo::T
+    hi::S
+end
```

Comment:
Maybe there should be an inner constructor checking that `L` and `R` are either `:closed` or `:open` ?